File: corecomply-mvp/api/CoreComply.Api/Domain/Entities/Report.cs
namespace CoreComply.Api.Domain.Entities;

public class Report
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty; // compliance, gap_analysis, audit, risk, executive
    public string Description { get; set; } = string.Empty;
    public DateTime? LastGenerated { get; set; }
    public string Status { get; set; } = "ready"; // ready, generating, scheduled, failed
    public string? Schedule { get; set; } // cron expression
    public string? Framework { get; set; }
    public string Format { get; set; } = "pdf"; // pdf, excel, csv
    public string CreatedBy { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public string? Parameters { get; set; } // JSON parameters
}

public class ReportTemplate
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Category { get; set; } = string.Empty; // compliance, audit, risk, executive
    public string? DefaultParameters { get; set; } // JSON
}

File: corecomply-mvp/web/src/types/report.ts
export interface Report {
  id: number;
  name: string;
  type: string;
  description: string;
  lastGenerated?: string | null;
  status: string;
  schedule?: string | null;
  framework?: string | null;
  format: string;
  createdBy: string;
  createdAt: string;
  parameters?: string | null;
}

export interface ReportTemplate {
  id: number;
  name: string;
  type: string;
  description: string;
  category: string;
  defaultParameters?: string;
}

export interface GenerateReportRequest {
  name: string;
  type: string;
  description?: string;
  format: string;
  framework?: string;
  parameters?: Record<string, any>;
}
